### PR TITLE
Discord presence - show the current game

### LINKED ISSUE #10720: [Feature Request]Expand Discord Integration with Rich Presence
I have no idea how to make a pull request, so that's why I made a issue instead:

Discord has created Rich Presence, which lets game creators expand how their games get detected, and what is shown in the rich presence feature. So, I've been wondering if that feature can have game detection implemented as in: The user status in Discord will still be "Playing PPSSPP". BUT, upon clicking them on the userlist, the Rich Presence shows, with the name of the game they are currently playing.

I feel like this would be soo much better than simply showing the user is using the emulator. But not the game.

https://discordapp.com/rich-presence

### PR DESCRIPTION
**Pull request by hrydgard**
Very basic integration with the popular chat client Discord, to show the game you're currently playing in the player status.

Fixes #10720

Unfortunately drags in yet another JSON library as a submodule.

### CODE REVIEW

File: UI/EmuScreen.cpp [hunk 1]
```diff
@@ -192,6 +193,10 @@ void EmuScreen::bootGame(const std::string &filename) {
 		g_Config.loadGameConfig(info->id);
 		// Reset views in case controls are in a different place.
 		RecreateViews();
+
+		g_Discord.SetPresenceGame(info->GetTitle().c_str());
+	} else {
+		g_Discord.SetPresenceGame("(unknown)");
```
Comment (unknownbrackets) [inline UI/EmuScreen.cpp]: Translate?  Maybe "unknown game" or "Untitled PSP Game" or something?

-[Unknown]

File: UI/DiscordIntegration.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		return;
+	
+	if (!initialized_) {
+		Init();
+	}
+
+#ifdef ENABLE_DISCORD
+	DiscordRichPresence discordPresence{};
+	discordPresence.state = gameTitle;
+	discordPresence.details = "Playing";
```
Comment (unknownbrackets) [inline UI/DiscordIntegration.cpp]: Should we translate this and "In menu"? Or does Discord?

-[Unknown]

File: UI/EmuScreen.cpp (hunk 1 above)
Comment (hrydgard) [reply UI/EmuScreen.cpp]: Did "Untitled PSP Game". Not sure which translation category makes the most sense though?

File: UI/DiscordIntegration.cpp (hunk 2 above)
Comment (hrydgard) [reply UI/DiscordIntegration.cpp]: Hm, good points. Translated both, but I don't think there's any way to set multi-language messages on Discord so it'll just be what the user uses.

File: UI/EmuScreen.cpp (hunk 1 above)
Comment (unknownbrackets) [reply UI/EmuScreen.cpp]: Screen sounds okay to me.  I guess we could create a separate one for Discord and eventual cloud saves or whatever, but doesn't seem that important...

-[Unknown]

File: ext/discord-rpc-build/discord-rpc.vcxproj [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    <ProjectGuid>{BEB0A821-3C7F-410F-A525-63AFBC69BF8F}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>discordrpc</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0.17134.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseDebugLibraries>true</UseDebugLibraries>
+    <PlatformToolset>v141</PlatformToolset>
```
Comment (unknownbrackets) [inline ext/discord-rpc-build/discord-rpc.vcxproj]: I think we need v140_xp or v141_xp (I don't know if the latter is a Real Thing, though)?

-[Unknown]

**Comment by unknownbrackets** [comment]
Hm.

> error MSB8020: The build tools for v141 (Platform Toolset = 'v141') cannot be found. To build using the v141 build tools, please install v141 build tools.

I'm guessing AppVeyor needs a configuration change to set the requirement to 2017?

-[Unknown]

**Comment by Amyrakunejo** [comment]
One query. Where do these files go? Just dropping them in with the main ppsspp directory does squat.
Unless, I have to restart Discord first...(?)

**Comment by hrydgard** [comment]
What files? This is included in all builds since this code was merged. There's nothing you need to do for it to work if you are using a recent enough build of PPSSPP.

**Comment by Amyrakunejo** [comment]
Since 1.6 was not running properly, I went back to 1.3.
I'm willing to guess that the version I use is incompatible then.

**Comment by hrydgard** [comment]
If a version, like 1.6, isn't running properly for you, please first try the latest (currently 1.7), then if that isn't working you should report the problem so that we can fix it. We had no support for Discord integration in 1.3 so it obviously won't work.

**Pull request merged**